app/controllers/concerns/authentication.rb

```diff
@@ -83,4 +83,17 @@ def identify_user_on_error_tracking_system
 
     Sentry.set_user(id: current_user.id)
   end
+
+  def current_impersonation
+    return unless impersonating?
+    
+    @current_impersonation ||= Impersonation.where(
+      user: current_user,
+      admin: true_user
+    ).where('created_at > ?', 24.hours.ago).last
+  end
+
+  def impersonating?
+    respond_to?(:true_user) && respond_to?(:current_user) && true_user && current_user && true_user != current_user
```

Comment:
```suggestion
    user_sign_in? && true_user != current_user
```

moreover add this method within `ApplicationController` with the following implementation

```ruby
def impersonating?
  false
end
```

in order to be sure this method is always implemented